### PR TITLE
add feature Master Passwords via Secrets Manager

### PR DESCRIPTION
**Pull request by ByJacob**
## what

- add parameter `manage_master_user_password`
- add parameter `master_user_secret_kms_key_id`

## why

- allow use new feature

## references

- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance#managed-master-passwords-via-secrets-manager-specific-kms-key
- https://github.com/hashicorp/terraform-provider-aws/blob/v4.61.0/CHANGELOG.md

### CODE REVIEW

File: outputs.tf
Comment (adamantike) [inline outputs.tf:51]: I think exposing the entire `master_user_secret` object would be useful for users, and will simplify this logic by avoiding those `local` values:

```hcl
output "master_user_secret" {
  value = one(aws_db_instance.default[*].master_user_secret)
  ```

**Comment by adamantike** [comment]
This would be really useful to have, and could potentially avoid the need for #118 for most users, while also not exposing the password in the state file. Can this be reviewed?

**Comment by adamantike** [comment]
@joe-niland, @Gowiem, gentle ping. Can this be reviewed?
It will be really useful to have this option merged, as this new feature gains more traction.

File: outputs.tf
Comment (Gowiem) [inline outputs.tf:54]:
```suggestion
output "master_user_secret" {
  value       = one(aws_db_instance.default[*].master_user_secret)
  description = "Secret object if configured with `var.database_manage_master_user_password = true`."
}
```

Comment (Gowiem) [reply outputs.tf:54]: This fixes up alignment + adds a slightly better description.

File: variables.tf
Comment (Gowiem) [inline variables.tf:51]: I feel like the `default` should be `false` instead of `null` because even though they should accomplish the same thing, `false` is more clear and keeps in line with `bool` arguments being binary. 
```suggestion
  default     = false
```

**Review by Gowiem — CHANGES_REQUESTED** [review]
@ByJacob this looks good and I'm happy to provide another review after my couple of small requests. One larger issue is that this module hasn't been updated in a while, and that has resulted in our tests rotting and them using an old version of our VPC module that includes a deprecated argument (see screenshot). Can you update the VPC + Subnet module usages in these two locations to the latest versions and then we'll re-run tests?

1. https://github.com/cloudposse/terraform-aws-rds/blob/main/examples/complete/main.tf#L5-L26
2. https://github.com/cloudposse/terraform-aws-rds/blob/main/examples/mssql/main.tf#L5-L24

CleanShot 2023-10-31 at 08 24 03

**Comment by Gowiem** [comment]
@adamantike thanks for the ping! I've seen you do that on a few PRs and it's appreciated when there are good fixes + enhancements like this one that fall through the cracks. Though I think we'd all like to do better, there are so many modules that the squeaky wheels get the grease. That said, I believe Erik + team are always looking for more maintainers so if you're interested in helping us get things reviewed + merged, please reach out to me or Erik via Slack and we can likely make that happen.

**Comment by adamantike** [comment]
@ByJacob, if needed, I can tackle the VPC and subnet module upgrade in the examples, for you to rebase your changes after that is done.

File: variables.tf
Comment (ByJacob) [reply variables.tf]: changed

**Comment by ByJacob** [comment]
> /terratest

@Gowiem I fixed tflint and Readme

**Comment by Gowiem** [comment]
@ByJacob Tests are failing on the following: 
CleanShot 2023-11-09 at 10 53 42

**Comment by ByJacob** [comment]
> @ByJacob Tests are failing on the following: CleanShot 2023-11-09 at 10 53 42

Id was change. I fix it and add identifier parameter

https://github.com/hashicorp/terraform-provider-aws/commit/a7cb14bfc969b896e6d7d1290529b572a016b46a

**Review by Gowiem — APPROVED** [review]
Solid work @ByJacob -- Thanks for working through the test failures!

**Comment by ByJacob** [comment]
> Solid work @ByJacob -- Thanks for working through the test failures!

It was enough to tighten the main branch, there it was fixed :D

**Comment by Gowiem** [comment]
@ByJacob release of your changes are blocked on some internal issue that I'm fixing in #164. Once that ships, a new release will be cut for `v1.1.0`

**Pull request merged**
